src/device/spu/noise.cpp

```diff
@@ -0,0 +1,42 @@
+#include "noise.h"
+#include <array>
+
+namespace {
+std::array<int8_t, 64> noise_addition = {
+	{1, 0, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1, 0, 1, 1, 0,
+	 1, 0, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1, 0, 1, 1, 0,
+	 0, 1, 1, 0, 1, 0, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1,
+	 0, 1, 1, 0, 1, 0, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1}};
+
+std::array<int8_t, 5> noise_halfcycle = {{0, 84, 140, 180, 210}};
```

Comment:
These values doesn't fit in 8 bit type - int16_t should be used instead.

Reply:
Will work with uint8_t too, fixed.